lib/active_hash/base.rb

```diff
@@ -157,14 +157,29 @@ def all(options={})
       def where(options)
         return @records if options.nil?
         (@records || []).select do |record|
-          options.all? { |col, match| record[col] == match }
+          match_options?(record, options)
         end
       end
 
       def find_by(options)
-        where(options).first
+        return all.first if options.nil?
```

Comment:
`find_by` looks to specification to return the `first` when arguments is `nil`.
https://github.com/zilkey/active_hash/blob/master/spec/active_hash/base_spec.rb#L268